pkg/patterns/declarative/reconciler.go

```diff
@@ -122,6 +130,24 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 	}
 	log.WithValues("objects", fmt.Sprintf("%d", len(objects.Items))).Info("built deployment objects")
 
+	if r.options.status != nil {
+		isValidVersion, err := r.options.status.VersionCheck(ctx, instance, objects)
+		if err != nil {
+			if !isValidVersion {
+				// r.client isn't exported so can't be updated in version check function
+				err := r.client.Status().Update(ctx, instance)
```

Comment:
So this actually "shadows" the err value we care about.  So on line 142 err will be nil.

Sometimes people use a different name e.g. `updateErr :=...`

But I think a better pattern - when you can use it - is to so `if err := r.client.Status()...; err != nil` as then the scope of the shadowing err should just be the `if` statement (I think!)